examples/pxScene2d/src/pxFileCache.cpp

```diff
@@ -0,0 +1,279 @@
+    mkdir(mDirectory.cString(), 0777);
+  }
+
+  return RT_OK;
+}
+
+rtError rtFileCache::cacheDirectory(rtString& dir)
+{
+  if (mDirectory.isEmpty())
+    return RT_ERROR;
+  dir = mDirectory;
+  return RT_OK;
+}
+
+rtError rtFileCache::removeData(const char* url)
+{
+  if (NULL == url)
+    return RT_ERROR;
+ 
+  rtString urlToRemove = url; 
+  rtString filename = getHashedFileName(urlToRemove);
+  if (! filename.isEmpty())
+  {
+    if (false == deleteFile(filename))
+    {
+      rtLogWarn("!!! deletion of cache failed for url(%s)",url);
+      return RT_ERROR;
+    }
+    mCurrentSize = mCurrentSize - mFileSizeMap[url];
+    mFileSizeMap.erase(url);
+  }
+  else
+  {
+    rtLogWarn("!!! Got wrong filehash for url during removal of data (%s)",url);
+    return RT_ERROR;
+  }
+  return RT_OK;
+}
+
+rtError rtFileCache::addToCache(const rtHttpCacheData& data)
+{
+  rtString url;
+  rtError  err = data.url(url);
+  if (RT_OK != err)
+    return err;
+
+  if  (url.isEmpty())
+    return RT_ERROR;
+
+  rtString filename =  getHashedFileName(url);
+  if (filename.isEmpty())
+  {
+    rtLogWarn("Problem in getting hash from the url(%s) while adding to cache ",url.cString());
+    return RT_ERROR;
+  }
+
+  bool ret = writeFile(filename,data);
+  if (true != ret)
+     return RT_ERROR;
+  int64_t fileSize = getFileSize(filename);
+  mCurrentSize += fileSize;
+  mFileSizeMap[url] = fileSize;
+  int64_t size = cleanup();
+  rtLogWarn("current size after insertion and cleanup (%d)",size);
+  return RT_OK;
+}
+
+rtError rtFileCache::getHttpCacheData(const char* url, rtHttpCacheData& cacheData)
+{
+  rtString urlToQuery = url;
+  rtString filename =  getHashedFileName(urlToQuery);
+  if (filename.isEmpty())
+  {
+    rtLogWarn("Problem in getting hash from the url(%s) while read from cache",url);
+    return RT_ERROR;
+  }
+  if (false == readFileHeader(filename,cacheData))
+    return RT_ERROR; 
+  return RT_OK;
+}
+
+void rtFileCache::clearCache() 
+{
+  string cmd = "rm -rf ";
+  cmd.append(mDirectory.cString());
+  cmd.append("/*");
+  system(cmd.c_str());
+  mFileSizeMap.clear();
+  mCurrentSize = 0;
+}
+
+int64_t rtFileCache::cleanup()
+{
+  if ( (mCurrentSize > mMaxSize) && !(mFileSizeMap.empty()))
+  {
+    map<rtString,int64_t>::iterator iter = mFileSizeMap.begin();
+   
+    do
+    {
+      rtString filename = getHashedFileName(iter->first);
+      if (! filename.isEmpty())
+      {
+        if(false == deleteFile(filename))
+        {
+          rtLogWarn("!!! deletion of cache failed during cleanup for url(%s)",iter->first.cString());
+        }
+        else
+        {
+	  mCurrentSize = mCurrentSize - iter->second;
+          mFileSizeMap.erase(iter);
+        }
+      }
+      iter++;
+    } while ((mCurrentSize > mMaxSize) && (iter != mFileSizeMap.end()));
+  }
+  return mCurrentSize;
+}
+
+rtString rtFileCache::getHashedFileName(const rtString& url)
+{
+  long int hash = hashFn(url.cString());
+  return to_string(hash).c_str();
```

Comment:
Change from to_string() to the below to support all platforms:

stringstream stream;
stream << hash;
return stream.str().c_str();